### PR TITLE
[spark] MigrateTableProcedure support keep origin table and rename target paimon table

### PR DESCRIPTION
**Pull request by xuzifu666**
Currently MigrateTableProcedure  only support migrate table with origin name，but some conditions user not want to drop origin table in hms or want to rename a new target table name，this pr is aim to support it.

Linked issue: close #xxx

### CODE REVIEW

File: paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/migrate/HiveMigrator.java [hunk 1]
```diff
@@ -78,6 +78,7 @@ public class HiveMigrator implements Migrator {
     private final String targetDatabase;
     private final String targetTable;
     private final Map<String, String> options;
+    private Boolean delete = true;
```
Comment (JingsongLi) [inline paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/migrate/HiveMigrator.java:81]: boolean

File: paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/procedure/MigrateTableProcedure.java [hunk 2]
```diff
@@ -77,9 +81,14 @@ public InternalRow[] call(InternalRow args) {
         String format = args.getString(0);
         String sourceTable = args.getString(1);
         String properties = args.isNullAt(2) ? null : args.getString(2);
+        Boolean deleteNeed = args.isNullAt(3) ? true : args.getBoolean(3);
```
Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/procedure/MigrateTableProcedure.java:84]: boolean

File: paimon-core/src/main/java/org/apache/paimon/migrate/Migrator.java [hunk 3]
```diff
@@ -24,4 +24,6 @@ public interface Migrator {
     void executeMigrate() throws Exception;
 
     void renameTable(boolean ignoreIfNotExists) throws Exception;
+
+    public void deleteOriginTable(Boolean delete) throws Exception;
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/migrate/Migrator.java:28]: boolean

File: docs/content/spark/procedures.md [hunk 4]
```diff
@@ -101,6 +101,18 @@ This section introduce all available spark procedures about paimon.
...
       </td>
     </tr>
+    <tr>
+      <td>migrate_table</td>
+      <td>
+         Migrate hive table to a paimon table. Arguments:
+            <li>source_type: the origin table's type to be migrated, such as hive. Cannot be empty.</li>
+            <li>table: name of the origin table to be migrated. Cannot be empty.</li>
+            <li>options: the table options of the paimon table to migrate.</li>
+            <li>delete_origin: whether to delete the origin table metadata from hms after migrate. Default is true</li>
```
Comment (JingsongLi) [inline docs/content/spark/procedures.md:111]: I think we should mention the option of target_table in the description. Without target_table, this option is difficult to understand.

Comment (xuzifu666) [reply docs/content/spark/procedures.md:111]: Get, <li>target_table: name of the target paimon table to migrate. If not set would keep the same name with origin hive table</li> this describtion would be OK？

File: paimon-core/src/main/java/org/apache/paimon/migrate/Migrator.java (hunk 3 above)
Comment (xuzifu666) [reply paimon-core/src/main/java/org/apache/paimon/migrate/Migrator.java]: addressed

File: paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/migrate/HiveMigrator.java (hunk 1 above)
Comment (xuzifu666) [reply paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/migrate/HiveMigrator.java:81]: addressedaddressed

File: paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/procedure/MigrateTableProcedure.java (hunk 2 above)
Comment (xuzifu666) [reply paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/procedure/MigrateTableProcedure.java]: addressed

**Pull request merged**